From ed2c6af506a4a9f1dfb8bf6b15d1949814c7df10 Mon Sep 17 00:00:00 2001
From: Simon Zolin <stsaz@yandex.ru>
Date: Wed, 29 Jun 2016 14:52:35 +0300
Subject: [PATCH] rename mpg123_init() and mpg123_decode()

---
 src/libmpg123/libmpg123.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/libmpg123/libmpg123.c b/src/libmpg123/libmpg123.c
index 26486df..f332c65 100644
--- a/src/libmpg123/libmpg123.c
+++ b/src/libmpg123/libmpg123.c
@@ -16,7 +16,7 @@
 
 static int initialized = 0;
 
-int attribute_align_arg mpg123_init(void)
+int attribute_align_arg _mpg123_init(void)
 {
 	if((sizeof(short) != 2) || (sizeof(long) < 4)) return MPG123_BAD_TYPES;
 
@@ -902,7 +902,7 @@ int attribute_align_arg mpg123_feed(mpg123_handle *mh, const unsigned char *in,
 	}
 */
 
-int attribute_align_arg mpg123_decode(mpg123_handle *mh, const unsigned char *inmemory, size_t inmemsize, unsigned char *outmemory, size_t outmemsize, size_t *done)
+int attribute_align_arg _mpg123_decode(mpg123_handle *mh, const unsigned char *inmemory, size_t inmemsize, unsigned char *outmemory, size_t outmemsize, size_t *done)
 {
 	int ret = MPG123_OK;
 	size_t mdone = 0;
-- 
2.4.3

